amp-pwa-reader/src/js/Article.js

```diff
@@ -41,6 +42,61 @@ class Article {
 
   }
 
+// see https://github.com/ampproject/amphtml/blob/master/spec/amp-shadow-doc.md
+// and https://jakearchibald.com/2016/fun-hacks-faster-content/
+// Recursively read and write chunks of data from the streaming API
+  stream() {
+    var shadowDoc = this.ampDoc;    // let's get this into the closure, and thus accessible to callback
+    var article = this;             // this too
+
+    fetch(this.proxyUrl).then(async response => {
+      let reader = response.body.getReader();
+      let decoder = new TextDecoder();
+
+      while (true) {
+        let chunk = await reader.read();
```

Comment:
nit, but would be interesting to go over the rest of the code and see what would change if we used await/async religiously.

Reply:
I just added this; to me, it makes the core streaming code much simpler.

I think it would also help to do it elsewhere, though probably not as drastically.